zap/src/main/java/org/parosproxy/paros/core/scanner/AbstractAppParamPlugin.java

```diff
@@ -137,6 +138,12 @@ protected void scan(List<NameValuePair> nameValuePairs) {
         }
     }
 
+    @Override
+    protected void sendAndReceive(HttpMessage message) throws IOException {
+        sendAndReceive(message, true);
+        variant.decodeResponseBody(message);
```

Comment:
Please, expose the protected method that allows to decode, otherwise it does not allow implementations to override this behaviour.

Reply:
ok i will make it this , maybe this will work 
`protected void decodeResponseBody(HttpMessage message) {}`

Reply:
Could you please clarify how to expose the protected decoding method? I'm unsure about implementation. Thanks!

Reply:
Like you suggested in a previous comment, it just needs to call `variant.decodeResponseBody(message);` (and have the `sendAndReceive` call the new method).

Reply:
sorry but still didn't get it, yeah sendAndReceive have to call variant.decodeResponseBody, which method i need to expose and what you mean by expose ?

Reply:
are you suggesting to write a new method like so that, subclasses of AbstractAppParamPlugin can use the decodeResponseBody method. 
```
protected void decodeResponseBody(HttpMessage message) {
    variant.decodeResponseBody(message);
}
```

Reply:
Yes.